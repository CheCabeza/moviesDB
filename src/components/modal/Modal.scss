@import '../../styles/colors';
@import '../../styles/generalStyles.scss';

*, *::after, *::before {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}
.Overlay {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba($color: #000000, $alpha: 0.7);
}

.MovieModal {
    position:fixed;
    top: 50%;
    left: 50%;
    width: 80%;
    height: 80%;
    border-radius: 30px;
    transform: translate(-50%, -50%);
    background-color: $yellow;
    padding: 50px;
    z-index: 1000;
    
    &__SimilarMovies {
        margin: 30px;
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;
        justify-content: center;

    }

    &__close-button {
        transform:  rotate(40 deg);
        border:none;
        background-color:transparent;
        outline:none;
        font-size:40px;
        position: absolute;
        top: 15px;
        right: 40px;
    }
    &__movie-details-section{
        margin: 10px;
        display: flex;
        flex-direction: row;
        border-radius: 20px;
        background-color: $blue;
        box-shadow: $shadow;
        img {
            border-radius: 20px 0px 0px 20px;
            width: 30vw;
            height: auto;
        }
        .movie-details-section__movie-details-info {
            padding: 50px;
        }
        p {
            margin: 20px;
        }
    }
    big {
        font-size: 30px;
        margin: 100px;
    }
}

.similarMoviesBox {
    width: 8vw;
    height: fit-content;
    margin: 20px;
    
    img {
        border-radius: 20px;
        width: 8vw;
        height: 12vw;
        box-shadow: $shadow;


    }
    button {
        transform:  rotate(40 deg);
        border:none;
        background-color:transparent;
        outline:none;
    }
    &__info {
        display: flex;
        flex-direction: row;
        justify-content: center;
        align-items: center;
        p {
            margin: 10px;

        }
    }
    :hover {
        cursor: pointer;
    }

}