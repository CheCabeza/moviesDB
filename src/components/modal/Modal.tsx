import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { RiCloseCircleFill } from 'react-icons/ri';
import { loadDetails, loadSimilar } from '../../redux/actions/actionCreator';
import { AppState } from '../../redux/reducers';
import './Modal.scss';

interface IModalProps {
  open: boolean;
  onClose: any;
  movieId: string;
  type: string;
}
interface IProduction {
  id: string;
  title?: string;
  name?: string;
  vote_average: string;
  poster_path: string;

}

export default function MovieModal({
  open, onClose, movieId, type,
}: IModalProps) {
  const details = useSelector((store: AppState) => store.details);
  const similarMovies = useSelector((store: AppState) => store.similar.results);
  const dispatch = useDispatch();
  const [movieID, setMovieID] = useState(movieId);

  useEffect(() => {
    dispatch(loadDetails(movieID, type));
    dispatch(loadSimilar(movieID, type));
  }, [movieID]);

  useEffect(() => {
    dispatch(loadDetails(movieId, type));
    dispatch(loadSimilar(movieId, type));
  }, [movieId]);

  if (!open) return null;

  return (
    <>
      <div className="Overlay" />
      <div className="MovieModal">
        <button type="button" onClick={onClose} className="MovieModal__close-button" aria-label="log out"><RiCloseCircleFill /></button>
        <div className="MovieModal__movie-details-section">
          <img src={`https://image.tmdb.org/t/p/w500${details.backdrop_path}`} alt={details.title} />
          <div className="movie-details-section__movie-details-info">
            {
              details.title ? (<h1>{details.title}</h1>) : <h1>{details.name}</h1>
            }

            <p>{details.overview}</p>
          </div>
        </div>

        <big>Similar Movies:</big>
        <div className="MovieModal__SimilarMovies">
          {
            similarMovies && similarMovies.slice(0, 6).map((element:IProduction) => (
              <div className="similarMoviesBox">
                <button type="button" onClick={() => { setMovieID(element.id); }}>
                  <img src={`https://image.tmdb.org/t/p/w500${element.poster_path}`} alt={element.title} />
                </button>
              </div>
            ))

          }
        </div>
      </div>
    </>
  );
}
