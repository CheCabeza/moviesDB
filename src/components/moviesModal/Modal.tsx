import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { RiCloseCircleFill } from 'react-icons/ri';
import { loadDetails, loadSimilar } from '../../redux/actions/actionCreator';
import './Modal.scss';

export default function MovieModal({
  open, onClose, movieId, type,
}: any) {
  const details = useSelector((store: any) => store.details);
  const similarMovies = useSelector((store: any) => store.similar.results);
  const dispatch = useDispatch();
  const [movieID, setMovieID] = useState(movieId);

  useEffect(() => {
    dispatch(loadDetails(movieID, type));
    dispatch(loadSimilar(movieID, type));
  }, [movieID]);

  useEffect(() => {
    dispatch(loadDetails(movieId, type));
    dispatch(loadSimilar(movieId, type));
  }, [movieId]);

  if (!open) return null;

  return (
    <>
      <div className="Overlay" />
      <div className="MovieModal">
        <button type="button" onClick={onClose} className="MovieModal__close-button" aria-label="log out"><RiCloseCircleFill /></button>
        <div className="MovieModal__movie-details-section">
          <img src={`https://image.tmdb.org/t/p/w500${details.backdrop_path}`} alt={details.title} />
          <div className="movie-details-section__movie-details-info">
            {
              details.title ? (<h1>{details.title}</h1>) : <h1>{details.name}</h1>
            }

            <p>{details.overview}</p>
          </div>
        </div>

        <p>Similar Movies:</p>
        <div className="MovieModal__SimilarMovies">
          {
            similarMovies && similarMovies.slice(0, 6).map((element:any) => (
              <div className="similarMoviesBox">
                <button type="button" onClick={() => { setMovieID(element.id); }}>
                  <img src={`https://image.tmdb.org/t/p/w500${element.poster_path}`} alt={element.title} />
                </button>
              </div>
            ))

          }
        </div>
      </div>
    </>
  );
}
