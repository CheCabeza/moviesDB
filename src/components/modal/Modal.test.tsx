import { Provider } from 'react-redux';
import React from 'react';
import { render } from '@testing-library/react';
import store from '../../redux/store/index';

import Modal from './Modal';

describe('Given a Details component', () => {
  test('Should render the Modal component', () => {
    const modalComponent = render(
      <Provider store={store({})}>
        <Modal open onClose={() => jest.fn} movieId="213" type="movie" />
      </Provider>,

    );
    expect(modalComponent).toMatchSnapshot();
  });
});
