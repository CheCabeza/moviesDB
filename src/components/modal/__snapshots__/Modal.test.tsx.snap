// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Details component Should render the Modal component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Overlay"
      />
      <div
        class="MovieModal"
      >
        <button
          aria-label="log out"
          class="MovieModal__close-button"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-11.414L9.172 7.757 7.757 9.172 10.586 12l-2.829 2.828 1.415 1.415L12 13.414l2.828 2.829 1.415-1.415L13.414 12l2.829-2.828-1.415-1.415L12 10.586z"
              />
            </g>
          </svg>
        </button>
        <div
          class="MovieModal__movie-details-section"
        >
          <img
            src="https://image.tmdb.org/t/p/w500undefined"
          />
          <div
            class="movie-details-section__movie-details-info"
          >
            <h1 />
            <p />
          </div>
        </div>
        <big>
          Similar Movies:
        </big>
        <div
          class="MovieModal__SimilarMovies"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Overlay"
    />
    <div
      class="MovieModal"
    >
      <button
        aria-label="log out"
        class="MovieModal__close-button"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-11.414L9.172 7.757 7.757 9.172 10.586 12l-2.829 2.828 1.415 1.415L12 13.414l2.828 2.829 1.415-1.415L13.414 12l2.829-2.828-1.415-1.415L12 10.586z"
            />
          </g>
        </svg>
      </button>
      <div
        class="MovieModal__movie-details-section"
      >
        <img
          src="https://image.tmdb.org/t/p/w500undefined"
        />
        <div
          class="movie-details-section__movie-details-info"
        >
          <h1 />
          <p />
        </div>
      </div>
      <big>
        Similar Movies:
      </big>
      <div
        class="MovieModal__SimilarMovies"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
